packages/lab/src/input-next/InputNext.tsx

```diff
@@ -0,0 +1,222 @@
+import { clsx } from "clsx";
+import {
+  AriaAttributes,
+  ChangeEvent,
+  ElementType,
+  FocusEvent,
+  FocusEventHandler,
+  forwardRef,
+  HTMLAttributes,
+  InputHTMLAttributes,
+  KeyboardEventHandler,
+  MouseEvent,
+  MouseEventHandler,
+  useRef,
+  useState,
+} from "react";
+import { makePrefixer, useControlled, useForkRef } from "@salt-ds/core";
+import { useFormFieldPropsNext } from "../form-field-context";
+
+import "./InputNext.css";
+
+const withBaseName = makePrefixer("saltInputNext");
+
+// TODO: Double confirm whether this should be extending DivElement given root is `<div>`.
+// And forwarded ref is not assigned to the root like other components.
+export interface InputProps
+  extends Omit<HTMLAttributes<HTMLDivElement>, "onChange" | "defaultValue"> {
+  /**
+   * The value of the `input` element, required for an uncontrolled component.
+   */
+  defaultValue?: HTMLInputElement["defaultValue"];
+  /**
+   * If `true`, the component is disabled.
+   */
+  disabled?: HTMLInputElement["disabled"];
+  /**
+   * The HTML element to render the Input, e.g. 'input', a custom React component.
+   */
+  inputComponent?: ElementType;
```

Comment:
i think we should remove `inputComponent`. @joshwooding do you have any info about why this was added initially?